sys/abi/src/lib.rs

```diff
@@ -598,6 +598,32 @@ impl core::convert::TryFrom<u32> for Sysnum {
     }
 }
 
+/// Representation of kipc numbers
+/// These get passed as an operation which is passed as a u16
+#[repr(u16)]
+pub enum Kipcnum {
+    ReadTaskStatus = 1,
+    RestartTask = 2,
+    FaultTask = 3,
+    ReadImageId = 4,
+    Reset = 5,
+}
+
+impl core::convert::TryFrom<u16> for Kipcnum {
```

Comment:
Recommending deriving `FromPrimitive` for this.

Reply:
I was following the convention of of defining it manually based on this comment for other functions

```
/// We're using an explicit `TryFrom` impl for `Sysnum` instead of
/// `FromPrimitive` because the kernel doesn't currently depend on `num-traits`
/// and this seems okay.
```